futures-util/src/future/join.rs

```diff
@@ -57,7 +59,7 @@ macro_rules! generate {
                 )*
 
                 if all_done {
-                    Poll::Ready(($(self.$Fut().take_output().unwrap_or_else(|| unreachable!())),*))
```

Comment:
This looks strange, but at the same time it feels like I'm missing something

Reply:
These aren't quite the same-- `unwrap` only works for `Debug` types, while `unwrap_or_else(|| unreachable!())` works for all types.

Reply:
Edit: ignore last-- that used to be true when this was a `Result`, but now that it is an `Option`, unwrapping is fine.